li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -417,6 +423,24 @@ public void close(long timeout, TimeUnit timeUnit) {
       }
 
       _partitionConsumerHighWatermark.computeIfAbsent(tp, _storedConsumerHighWatermark)._currentConsumerHighWatermark = consumerRecord.offset();
+      LargeMessageHeaderValue largeMessageHeaderValue = null;
+      Headers headers = new RecordHeaders(consumerRecord.headers());
+      Iterator<Header> iterable = headers.headers(Constants.LARGE_MESSAGE).iterator();
+      if (iterable.hasNext()) {
+        Header header = iterable.next();
+        largeMessageHeaderValue = LargeMessageHeaderValue.fromBytes(header.value());
+      }
+      // Once the large message is parsed, remove any such key from record headers
+      headers.remove(Constants.LARGE_MESSAGE);
+      if (largeMessageHeaderValue != null) {
+        largeMessageHeaderValue = new LargeMessageHeaderValue(
+            LargeMessageHeaderValue.Type.LM_ENCRYPTION_AVRO,
+            largeMessageHeaderValue.getUuid(),
+            -1,
```

Comment:
What is the reason for wiping out the sequence id?

Reply:
Because the large message will not have any sequence number associated with it. It is only the internal splits that will have segment ids starting from zero. This is the record that will be returned to the end-user.

Reply:
Wouldn't the same logic apply for `getNumberOfSegments` ? I mean it just feels like we are exposing the internal details unintentionally, if we are exposing the  idea of segments.

Reply:
The number of segments is required for the auditing use case(i.e,. count the actual number of messages by using a Likafka client). A case can be made that UUID isn't required but since we are using a certain header for large message -- might as well use the same.

Reply:
>  A case can be made that UUID isn't required but since we are using a certain header for large message -- might as well use the same. 
I am not following the line of logic here. I would rather not expose anything internal as much as possible. That's why I am concerned about exposing the number of segments. It seems to serve the purpose for auditing. I don't see any use-case for exposing uuid.

Reply:
Set the UUID to an empty value.